chingu-voyages/v52-tier3-team-38
# Ross/feature/resident navbar

# Description
This PR implements a comprehensive refactor of the user interface components, focusing on improving navigation, profile display, and overall user experience. The changes consolidate navigation components, implement dynamic routing, and enhance the user profile page with appointment management functionality.

## Related to the following Jira/Trello issue 

- Resident Interface.
- Navigation Bar
- Responsiveness

## Main changes explained:
- Consolidated UserHeader and UserNavbar into a single, more efficient UserNavbar component.
- Implemented dynamic user routing in UserNavbar for improved navigation.
- Enhanced user profile page to display user information and user-specific appointments.
- Refactored navigation system to use Bootstrap React Navbar and NavDropdown for better responsiveness.
- Improved mobile and desktop views with responsive icons and text.
- Removed deprecated ServiceHistory component and associated links.
- Implemented "use client" directive for proper Next.js client-side rendering.
- Resolved multiple merge conflicts with development branch to ensure code consistency.

## How to test:
1. Check out the current branch.
2. Run npm install and npm run dev to run this PR locally.
3. Navigate to /user/profile to view the updated user interface. Some user info, and an appointment (maybe?) should appear.
4. Verify that the consolidated navbar appears correctly on both desktop and mobile views.
5. Test the responsive design by resizing the browser window.
6. Confirm that user-specific appointments are displaying correctly.
7. Verify all navigation links are working as expected.


## Screenshots or videos of changes:

User Navbar Mobile View
![image](https://github.com/user-attachments/assets/6a0b123f-44a9-4955-89fb-c6d3d3540214)

![image](https://github.com/user-attachments/assets/3e697696-9692-42f8-a143-6d2418f2a731)

User Navbar Desktop View
![image](https://github.com/user-attachments/assets/793d045d-a250-4dde-aefc-663e1d6871d1)

![image](https://github.com/user-attachments/assets/845623a9-82f4-457b-b777-a1169abae828)

User Profile Page -- No appointment
![image](https://github.com/user-attachments/assets/1f942041-fce5-4b91-9a46-86073674f386)

User Profile Page -- Installation Request
![image](https://github.com/user-attachments/assets/d1523f77-1408-4f36-b046-8092a4ec4b28)



## Note:
- More styling will be done as the unauthenticated and admin navbars are worked on. Some pages still have to be styled.
- The new UserNavbar component now replaces the previous separate header and footer navbar components.
- This PR combines features from multiple branches (user-navigation, resident-profile-page) and other merges.
- Mobile responsiveness has been prioritized with the latest commits.
- Bootstrap React components are now being used for more consistent styling and behavior.
